janitor/functions/pivot.py

```diff
@@ -275,21 +318,24 @@ def _data_checks_pivot_longer(
             names_to = [*names_to]
         check("names_to", names_to, [list])
 
-        unique_names_to = set()
+        uniques = set()
         for word in names_to:
-            if not isinstance(word, str):
-                raise TypeError(
-                    "All entries in the names_to argument must be strings. "
-                    f"{word} is of type {type(word).__name__}."
-                )
-
-            if (word in unique_names_to) and (word != ".value"):
+            check(f"{word} in names_to", word, [str])
+            if (word in uniques) and (word != ".value"):
                 raise ValueError(f"{word} already exists in names_to.")
-            unique_names_to.add(word)  # noqa: PD005
-        unique_names_to = None
+            uniques.add(word)  # noqa: PD005
```

Comment:
```suggestion
            uniques.add(word)
```
PD005 ("Use arithmetic operator instead of method") is probably no longer relevant anymore here, let's remove it and see if the linter complains?